src/hotspot/share/opto/matcher.cpp

```diff
@@ -2500,6 +2500,13 @@ void Matcher::find_shared_post_visit(Node* n, uint opcode) {
       n->del_req(3);
       break;
     }
+    case Op_PartialSubtypeCheck: {
```

Comment:
This could use a comment.  Changing the shape with BinaryNode is standard, but duplicating inputs seems to be something new.

Reply:
Good point. Does it look better?
```
      if (UseSecondarySupersTable && n->in(2)->is_Con()) {
        // PartialSubtypeCheck uses both constant and register operands for superclass input.
        n->set_req(2, new BinaryNode(n->in(2), n->in(2)));
        break;
```